requirements_build.txt

```diff
@@ -1,2 +1,3 @@
 cython
+setuptools<50.0
```

Comment:
I agree that `<` is probably not a good idea.

Reply:
Currently, looks it's unavoidable if you'd like to build scikit-learn from sources, unfortunately: https://github.com/numpy/numpy/pull/20640#issuecomment-999800999.
> By the way, y'all really should pin `setuptools` in your release branches. This is not done currently (https://github.com/scikit-learn/scikit-learn/blob/1.0.X/pyproject.toml#L4), and hence the breakage you're seeing in your CI will also occur when installing from an sdist with `pip install scikit-learn`.


Anyway, we are going to have a big refactoring of the package building routine with Python 3.12 release according to https://github.com/scikit-learn/scikit-learn/issues/21499.